From 43752126e6fece198dbcd5449640c13e4baa7b21 Mon Sep 17 00:00:00 2001
From: Eric Bentley <eric.bentley@lairdtech.com>
Date: Wed, 27 Aug 2014 10:18:56 -0400
Subject: [PATCH] x86 compiler bit shifts signed chars differently from arm

Bit shifting signed bytes with the x86 compiler results in 0xff values
for the shifted element.  This does not occur with the arm compiler.
The issue does not occur when using unsigned char elements.

BUG: 6054
---
 drivers/net/wireless/ath/ath6kl/core.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/wireless/ath/ath6kl/core.h b/drivers/net/wireless/ath/ath6kl/core.h
index 9774fba..f8b9345 100644
--- a/drivers/net/wireless/ath/ath6kl/core.h
+++ b/drivers/net/wireless/ath/ath6kl/core.h
@@ -852,7 +852,7 @@ struct ath6kl {
 		struct wmi_set_htcap_cmd htcap_params_2ghz;
 		struct wmi_set_htcap_cmd htcap_params_5ghz;
 		char AP_Name[17];
-		char AP_IP[4];
+		unsigned char AP_IP[4];
 	} laird;
 };
 
-- 
1.7.9.5

